### PR TITLE
Release 21.05

### PR DESCRIPTION
**Pull request by jonringer**
Attempt to bump materials for 21.05.

nixUnstable doesn't want to build, and nix (stable) doesn't build with `utillinuxMinimal` being aliased.

### CODE REVIEW

**Comment by jonringer** [comment]
cc @garbas

**Comment by samueldr** [comment]
cc @edolstra too for the Nix weirdness

**Comment by jonringer** [comment]
I'm able to build `nixUnstable` using my server, which is running 2.3.11. But not my desktop, which is using `nixUnstable`. And obviously flakes are meant to be used with nixUnstable

**Comment by jonringer** [comment]
The actual error is:
```
 > autoreconf: running: aclocal --force
 > aclocal: error: cannot open /nix/store/1hbdwchm6mkqhn8ygdgjjj9kkmps8ilv-autoconf-2.71/bin/autom4te --language Autoconf-without-aclocal-m4 /nix/store/9mvzczd9dpp2yhjkg2cxcc3x7snpl6z1-automake-1.16.3/share/aclocal-1.16/internal/ac-config-macro-dirs.m4 '/nix/store/x7qywzjq61hh3agc101hc3am88xp7vd2-autoconf-archive-2021.02.19/share/aclocal/ax_boost_base.m4' '/nix/store/b8mzg1b1s3lh2zvmz1yichqprzhi0f2d-pkg-config-wrapper-0.29.2/share/aclocal/pkg.m4' --trace='AC_DEFUN:$f::$n::${::}%' --trace='AC_DEFUN_ONCE:$f::$n::${::}%' --trace='AU_DEFUN:$f::$n::${::}%' --trace='_AM_AUTOCONF_VERSION:$f::$n::${::}%' --trace='AC_CONFIG_MACRO_DIR_TRACE:$f::$n::${::}%' --trace='AC_CONFIG_MACRO_DIR:$f::$n::${::}%' --trace='_AM_CONFIG_MACRO_DIRS:$f::$n::${::}%' --trace='PKG_CHECK_VAR:$f::$n' --trace='PKG_NOARCH_INSTALLDIR:$f::$n' --trace='AX_BOOST_BASE:$f::$n' --trace='PKG_PROG_PKG_CONFIG:$f::$n' --trace='PKG_CHECK_EXISTS:$f::$n' --trace='PKG_CHECK_MODULES_STATIC:$f::$n' --trace='PKG_INSTALLDIR:$f::$n' --trace='_AX_BOOST_BASE_RUNDETECT:$f::$n' --trace='PKG_CHECK_MODULES:$f::$n' --trace='_PKG_SHORT_ERRORS_SUPPORTED:$f::$n' --trace='_AX_BOOST_BASE_TONUMERICVERSION:$f::$n' --trace='NEED_PROG:$f::$n' configure.ac |: No such file or directory
       > autoreconf: error: aclocal failed with exit status: 1
```
on 2.3.11:
```
autoreconf: running: aclocal --force
autoreconf: configure.ac: tracing
autoreconf: configure.ac: not using Libtool
```

**Comment by jonringer** [comment]
could be related? https://github.com/NixOS/nix/issues/4810

**Comment by jonringer** [comment]
Also, related: https://github.com/NixOS/nix/pull/4863

**Comment by Kloenk** [comment]
I'm not sure I understand the problem. How can I reproduce this build failure? nixUnstable (nixpkgs/nixos-21.05#nixUnstable) does build on my x86_64-linux (with nix 2.4pre20210527_f10d302). Do I have to checkout anything else, to reproduce this? 

nixUnstable aarch64-darwin does not build, but this is because of the lowdown sandbox (see https://github.com/NixOS/nix/pull/4865)

File: flake.nix [hunk 1]
```diff
@@ -6,9 +6,9 @@ rec {
 
   # These inputs are used for the manuals and release artifacts
   inputs.released-nixpkgs-unstable = { url = "nixpkgs/nixos-unstable"; };
-  inputs.released-nixpkgs-stable = { url = "nixpkgs/nixos-20.09"; };
+  inputs.released-nixpkgs-stable = { url = "nixpkgs/nixos-21.05"; };
   inputs.released-nix-unstable = { url = "github:nixos/nix/master"; };
-  inputs.released-nix-stable = { url = "github:nixos/nix/latest-release"; flake = false; };
+  inputs.released-nix-stable = { url = "github:jonringer/nix?ref=fix-2.3-build-stable"; flake = false; };
```
Comment (davidak) [inline flake.nix:11]: shouldn't this be an official release instead of `github:jonringer`?

Comment (jonringer) [reply flake.nix:11]: it should, https://github.com/NixOS/nix/pull/4863 wasn't merged when I created the PR

**Comment by jonringer** [comment]
> How can I reproduce this build failure? nixUnstable (nixpkgs/nixos-21.05#nixUnstable) does build on my x86_64-linux (with nix 2.4pre20210527_f10d302). Do I have to checkout anything else, to reproduce this?

how did you get 20210527? on master:
```
[09:58:41] jon@jon-desktop /home/jon/projects/nixpkgs (master)
$ git pull
Already up to date.
[10:02:08] jon@jon-desktop /home/jon/projects/nixpkgs (master)
$ nix eval .#nixUnstable.version
"2.4pre20210503_6d2553a"
```

**Comment by garbas** [comment]
Closing the PR in favor of #718 (to have a preview)

**Comment by Kloenk** [comment]
> > How can I reproduce this build failure? nixUnstable (nixpkgs/nixos-21.05#nixUnstable) does build on my x86_64-linux (with nix 2.4pre20210527_f10d302). Do I have to checkout anything else, to reproduce this?
> 
> 
> 
> how did you get 20210527? on master:
> 
> ```
> 
> [09:58:41] jon@jon-desktop /home/jon/projects/nixpkgs (master)
> 
> $ git pull
> 
> Already up to date.
> 
> [10:02:08] jon@jon-desktop /home/jon/projects/nixpkgs (master)
> 
> $ nix eval .#nixUnstable.version
> 
> "2.4pre20210503_6d2553a"
> 
> ```


This could be a commit from this pr, which never got pushed https://github.com/NixOS/nix/pull/4865. but as this pr only enables aarch64-darwin. It should not fixe any errors you have.

File: flake.nix (hunk 1 above)
Comment (jonringer) [reply flake.nix]: This was fixed

**Pull request closed without merging**